internal/crypto/cert.go

```diff
@@ -59,3 +65,48 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// CompleteCertificateChain returns the complete certificate chain by appending the parent certificate from the certificate bundle to the original certificate chain.
+// If the certificate bundle path is not set in the pluginConfig, it returns the original certificate chain.
+// If an error occurs while reading the certificate bundle or parsing its PEM, it returns a nil slice and a non-nil error indicating the failure reason.
+func CompleteCertificateChain(pluginConfig map[string]string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
+	certBundlePath, ok := pluginConfig[certBundleKey]
+	// if certbundle is not set, return the original certs
+	if !ok {
+		return originalCerts, nil
+	}
+	certBundlePath = strings.Trim(certBundlePath, " ")
+
+	certBundleBytes, err := os.ReadFile(certBundlePath)
+	if err != nil {
+		return nil, err
+	}
+
+	certBundle, err := parsePEM(certBundleBytes)
```

Comment:
Just an open discussion, what if the user provided cert bundle file is in DER format? In this case, `block, rest := pem.Decode(data)` would have `block == nil`. Should we have something like:
```
block, rest := pem.Decode(data)
if block == nil {
  // data may be in DER format
  derCerts, err := x509.ParseCertificates(data)
  if err != nil {
	  return nil, err
  }
  certs = append(certs, derCerts...)
} else {
  // data is in PEM format
  for block != nil {
	  cert, err := x509.ParseCertificate(block.Bytes)
	  if err != nil {
		  return nil, err
	  }
	  certs = append(certs, cert)
	  block, rest = pem.Decode(rest)
  }
}
```

Reply:
We can keep it simple now according to today's meeting, so just support PEM format for this PR.